packages/cardano-services/src/ChainHistory/BlockrostChainHistoryProvider/BlockfrostChainHistoryProvider.ts

```diff
@@ -87,65 +107,101 @@ export class BlockfrostChainHistoryProvider extends BlockfrostProvider implement
   }
 
   protected async fetchPoolRetireCerts(hash: string): Promise<WithCertIndex<Cardano.PoolRetirementCertificate>[]> {
-    const response = await this.blockfrost.txsPoolRetires(hash);
-    return response.map(({ pool_id, retiring_epoch, cert_index }) => ({
-      __typename: Cardano.CertificateType.PoolRetirement,
-      cert_index,
-      epoch: Cardano.EpochNo(retiring_epoch),
-      poolId: Cardano.PoolId(pool_id)
-    }));
+    return this.blockfrost.txsPoolRetires(hash).then((response) =>
+      response.map(({ pool_id, retiring_epoch, cert_index }) => ({
+        __typename: Cardano.CertificateType.PoolRetirement,
+        cert_index,
+        epoch: Cardano.EpochNo(retiring_epoch),
+        poolId: Cardano.PoolId(pool_id)
+      }))
+    );
   }
 
   protected async fetchPoolUpdateCerts(hash: string): Promise<WithCertIndex<Cardano.PoolRegistrationCertificate>[]> {
-    const response = await this.blockfrost.txsPoolUpdates(hash);
-    return response.map(({ pool_id, cert_index }) => ({
-      __typename: Cardano.CertificateType.PoolRegistration,
-      cert_index,
-      poolId: Cardano.PoolId(pool_id),
-      poolParameters: ((): Cardano.PoolParameters => {
-        this.logger.warn('Omitting poolParameters for certificate in tx', hash);
-        return null as unknown as Cardano.PoolParameters;
-      })()
-    }));
+    return this.blockfrost.txsPoolUpdates(hash).then((response) =>
+      response.map(({ pool_id, cert_index, fixed_cost, margin_cost, pledge, reward_account, vrf_key }) => ({
+        __typename: Cardano.CertificateType.PoolRegistration,
+        cert_index,
+        poolId: Cardano.PoolId(pool_id),
+        poolParameters: {
+          cost: BigInt(fixed_cost),
+          id: pool_id as Cardano.PoolId,
+          margin: Cardano.FractionUtils.toFraction(margin_cost),
+          owners: [],
+          pledge: BigInt(pledge),
+          relays: [],
+          rewardAccount: reward_account as Cardano.RewardAccount,
+          vrf: vrf_key as Cardano.VrfVkHex
+        }
+      }))
+    );
+  }
+
+  async fetchCBOR(hash: string): Promise<string> {
```

Comment:
just nitpicking proposal: use the `TransactionId` type instead of `string`

Reply:
I tried... but this code is based on Blockfrost client package, not on our core types: doing that would require to add a lot of type casts.